public/index.php

```diff
@@ -23,17 +23,31 @@
     (new Dotenv())->load(__DIR__.'/../.env');
 }
 
-if ($_SERVER['APP_DEBUG'] ?? ('prod' !== ($_SERVER['APP_ENV'] ?? 'dev'))) {
+$env = $_SERVER['APP_ENV'] ?? 'dev';
+$debug = $_SERVER['APP_DEBUG'] ?? ('prod' !== $env);
+
+if ($debug) {
     umask(0000);
 
     Debug::enable();
 }
 
-// This is enabled to allow for correctly handling HTTPS on Heroku
-Request::setTrustedProxies(['0.0.0.0/0'], Request::HEADER_FORWARDED);
-
-$kernel = new Kernel($_SERVER['APP_ENV'] ?? 'dev', $_SERVER['APP_DEBUG'] ?? ('prod' !== ($_SERVER['APP_ENV'] ?? 'dev')));
+$kernel = new Kernel($env, $debug);
 $request = Request::createFromGlobals();
+
+// This is enabled to allow for correctly handling HTTPS on Heroku:
+// https://devcenter.heroku.com/articles/getting-started-with-symfony#trusting-the-load-balancer
+// The code snippet in the Heroku docs is for the Symfony 3 branch;
+// the way it worked was by adding headers to a blacklist by calling
+// Request::setTrustedHeaderName($name, $value) with $value set to null.
+// In Symfony 4, Request::setTrustedProxies() takes a second parameter,
+// which defines a whitelist, so instead of removing the headers we don't want to trust,
+// we explicitly state a bit field of which headers can be trusted.
+Request::setTrustedProxies(
+    [$request->server->get('REMOTE_ADDR')],
+    Request::HEADER_X_FORWARDED_FOR ^ Request::HEADER_X_FORWARDED_PROTO ^ Request::HEADER_X_FORWARDED_PORT
```

Comment:
This works but a shorter (and IMHO clearer) way to express this would be `Request:: HEADER_X_FORWARDED_ALL  ^ HEADER_X_FORWARDED_HOST`